lib/IRGen/DebugTypeInfo.cpp

```diff
@@ -88,10 +89,17 @@ DebugTypeInfo DebugTypeInfo::getArchetype(swift::Type Ty, llvm::Type *StorageTy,
                                           Size size, Alignment align) {
   DebugTypeInfo DbgTy(Ty.getPointer(), StorageTy, size,
                       align, true, true);
+  assert(StorageTy && "StorageType is a nullptr");
   assert(!DbgTy.isContextArchetype() && "type metadata cannot contain an archetype");
   return DbgTy;
 }
 
+DebugTypeInfo DebugTypeInfo::getForwardDecl(swift::Type Ty) {
+  DebugTypeInfo DbgTy(Ty.getPointer(), nullptr, {}, Alignment(1), true,
```

Comment:
Wdyt of making `StorageType` private in DebugTypeInfo and add a getter for it? This could assert if the field is null _and_ the size is known?

Reply:
Done in the second commit.